lib/minidoc/duplicate_key.rb

```diff
@@ -0,0 +1,13 @@
+class Minidoc::DuplicateKey < Mongo::OperationFailure
+  DUPLICATE_KEY_ERROR_CODE = 11000
+
+  def self.duplicate_key_exception(exception)
+    if exception.respond_to?(:error_code) &&
+      exception.error_code == DUPLICATE_KEY_ERROR_CODE
```

Comment:
this formatting is hard to grok visually, because the second part of the `if` blends together with the line below. single line